internal/service/searchindex/resource_search_index.go

```diff
@@ -281,9 +281,7 @@ func resourceUpdate(ctx context.Context, d *schema.ResourceData, meta any) diag.
 				"cluster_name": clusterName,
 				"index_id":     indexID,
 			}))
-			resourceDelete(ctx, d, meta)
-			d.SetId("")
-			return diag.FromErr(fmt.Errorf("error creating index in cluster (%s): %s", clusterName, err))
+			return diag.FromErr(fmt.Errorf("error updating index in cluster (%s). mongodbatlas_search_index resource was not deleted : %s", clusterName, err))
```

Comment:
I wonder why do we need to specify the resource was not deleted?

Reply:
This is mostly for users who might be expecting the previous incorrect behavior of the index being deleted in case of errors/timeouts, see [comment](https://jira.mongodb.org/browse/CLOUDP-229877?focusedCommentId=6100097&focusedId=6100097&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-6100097)